# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.h
# Commit: 0347d8673dee
# Full Hash: 0347d8673deea522ae15120fe89e2552bf822a6d
# Author: Emily McDonough <emcdonough@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1800546
# File Overlap Count: 1
# Description:
#   Bug 1800546 Part 1 - Use the style given the first page name for setting default orientation when printing r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D162058
# ==============================================================================

diff -r 0cda34ba7391 -r 0347d8673dee layout/printing/nsPrintJob.h
--- a/layout/printing/nsPrintJob.h	Sat Dec 03 01:15:55 2022 +0200
+++ b/layout/printing/nsPrintJob.h	Fri Dec 02 23:22:50 2022 +0000
@@ -8,6 +8,7 @@
 
 #include "mozilla/Attributes.h"
 #include "mozilla/layout/RemotePrintJobChild.h"
+#include "mozilla/Maybe.h"
 #include "mozilla/UniquePtr.h"
 
 #include "nsCOMPtr.h"
@@ -283,6 +284,11 @@
 
   int32_t mNumPrintablePages = 0;
 
+  // Indicates if the page has a specific orientation from @page { size }.
+  // Stores true if this is landscape, false if this is portrait, or nothing
+  // if there is no page-size-orientation.
+  mozilla::Maybe<bool> mMaybeCSSPageLandscape;
+
   // If true, indicates that we have started Printing but have not gone to the
   // timer to start printing the pages. It gets turned off right before we go
   // to the timer.